android/src/main/java/com/polidea/flutter_ble_lib/event/ScanningStreamHandler.java

```diff
@@ -22,9 +25,33 @@ public void onCancel(Object o) {
         scanResultsSink = null;
     }
 
-    public void onScanResult(ScanResult scanResult) throws JSONException {
+    public void onScanResult(ScanResult scanResult) {
         if (scanResultsSink != null) {
-            scanResultsSink.success(scanResultJsonConverter.toJson(scanResult));
+            try {
+                scanResultsSink.success(scanResultJsonConverter.toJson(scanResult));
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public void onError(BleError error) {
```

Comment:
All of these methods should be `synchronized` - there could be a race condition on cancelling the stream and pushing new value to it.